File: 2913-subarrays-distinct-element-sum-of-squares-i/2913-subarrays-distinct-element-sum-of-squares-i.py
class Solution:
    def sumCounts(self, nums: List[int]) -> int:
        ans = []
        n = len(nums)
        for i in range(n):
            for j in range(i, n+1):
                ans.append(len(set(nums[i:j]))**2)
                
        return sum(ans)

File: 2913-subarrays-distinct-element-sum-of-squares-i/2913-subarrays-distinct-element-sum-of-squares-i.rs
impl Solution {
    pub fn sum_counts(nums: Vec<i32>) -> i32 {
        let mut total = 0;
        let n = nums.len();
        
        for i in 0..n {
            for j in i..n {
                let sa = &nums[i..=j];
                let dc = sa.iter().collect::<std::collections::HashSet<_>>().len();
                total += (dc * dc) as i32;
            }
        }
        
        total
    }
}